agent/tool-scripts/prometheus-metrics

```diff
@@ -160,19 +139,39 @@ case "$mode" in
 	which go &>/dev/null
 	if [[ $? != 0 ]]; then
 		yum -y install go &>/dev/null
+		if [[ $? != 0 ]]; then
+			error_log "[$script_name] \"yum -y install go\" failed!"
+			exit 1
+		fi
 	fi
 	mkdir -p $pbench_install_dir/.go
+	if [[ $? != 0 ]]; then
+		error_log "[$script_name] \"mkdir -p $pbench_install_dir/.go\" failed!"
+		exit 1
+	fi
 	GOPATH=$pbench_install_dir/.go go get github.com/prometheus/prom2json/cmd/prom2json
+	if [[ $? != 0 ]]; then
+		error_log "[$script_name] \"go get github.com/prometheus/prom2json/cmd/prom2json\" failed!"
+		exit 1
+	fi
 	;;
 	start)
 	mkdir -p $tool_output_dir/json
+	if [[ $? != 0 ]]; then
+		error_log "[$script_name] \"mkdir -p $tool_output_dir/json\" failed!"
+		exit 1
+	fi
 	while read -u 9 file; do
-		# If the port, cert or key is not set while registering the tool, read the input file for host, port, cert and key if any. 
-		# In case the file doesn't have certs, port and they are not defined while registering the tool, the default values are set.
-		# host port=<port> cert=<cert> key=<key>
+		# If the port, cert or key is not set while registering
```

Comment:
Given you've pulled the `--port,--cert and --key` opts doesn't it have to be read from the input file exclusively?

Reply:
They were not being used at all.  If they were given as values in the inventory file, then they were not used the default values.

If you think we should fix that, I'll add them back and make it so that the defaults are used.

Reply:
I would just remove/edit the comment as the `port, cert or key is not set while registering` can't be used while registering now.